### PR TITLE
Task 8694r3zk8 - Post-Display-Counter plus technical views

### PR DESCRIPTION
**Pull request by tgrycuk**
This pull request introduces a feature to count and record the number of views each post receives, along with technical views.

### CODE REVIEW

File: core/views.py [hunk 1]
```diff
@@ -1 +1,39 @@
...
+
+    template_name = "post-detail.html"
+
+    def get_context_data(self: "PostDetailView", **kwargs: dict) -> dict:
+        context = super().get_context_data(**kwargs)
+
+        post_id = self.kwargs.get("pk")
+        post = get_object_or_404(Post, pk=post_id)
+        post.view_count += 1
+        post.save(update_fields=["view_count"], skip_version_check=True)
```
Comment (jacoor) [inline core/views.py]: I am thinking it should be update as here: 
https://github.com/pymasterspl/reddit/blob/fe257582955c20ea67e95b781a348c43d0d6785e/users/models.py#L53-L54
Reasoning: whenever save is run, all signals are run - post saves and all this that django runs. We don't need that, we only want to bump +1 to the read count. 
So please change to update.

Comment (jacoor) [inline core/views.py]: This should be a method on post itself, not in the view. The method itself is ok, but really, should be in model - after all it's logic on the model. 
(lines 31 to 34)

Then, the call to post.update_display_counter() should be most likely called in view get_object method, so tou will not even need get_context_data.

File: core/models.py [hunk 2]
```diff
@@ -78,12 +78,14 @@ class Post(GenericModel):
         max_length=32,
         help_text="Hash of the title + content to prevent overwriting already saved post",
     )
+    view_count = models.IntegerField(default=0)
 
     def __str__(self: "Post") -> str:
         return f"@{self.author}: {self.title}"
 
     def save(self: "Post", *args: int, **kwargs: int) -> None:
-        if self.pk is not None and self.generate_version() == self.version:
```
Comment (jacoor) [inline core/models.py]: This change won't be needed, please check below.

**Review by jacoor — CHANGES_REQUESTED** [review]

File: core/views.py (hunk 1 above)
Comment (swietlikm) [reply core/views.py]: You can try to use F() expression that is great for db performance
[https://docs.djangoproject.com/en/5.0/ref/models/expressions/#f-expressions](url)

File: core/models.py (hunk 2 above)
Comment (jacoor) [reply core/models.py]: Please revert the changes to save. Not needed now.

File: core/models.py [hunk 3]
```diff
@@ -127,6 +131,9 @@ def vote(self: "Post", user: User, choice: str) -> None:
     def get_images(self: "Post") -> models.QuerySet:
         return Image.objects.filter(post=self)
 
+    def update_display_counter(self: "Post") -> None:
+        Post.objects.filter(pk=self.pk).update(display_counter=F("display_counter") + 1)
```
Comment (jacoor) [inline core/models.py:133]: Nice use of F.

**Review by jacoor — CHANGES_REQUESTED** [review]

**Review by jacoor — APPROVED** [review]

**Pull request merged**